infrastructure/httpapi/handlers/validators.go

```diff
@@ -285,14 +303,56 @@ func (handler *Validators) getAverageBlockTime() (*big.Float, error) {
 		}
 	}
 
-	totalBlockTimeMilliSecond := new(big.Float).Quo(
-		new(big.Float).SetInt(totalBlockTime),
-		new(big.Float).SetInt64(int64(1000000)),
-	)
-	averageBlockTimeMilliSecond := new(big.Float).Quo(
-		totalBlockTimeMilliSecond,
-		new(big.Float).SetInt(totalBlockCount),
-	)
+	nRecentBlocks := big.NewInt(nRecentBlocksInInt)
+
+	isTotalBlockCountExcess := (totalBlockCount.Cmp(nRecentBlocks) == 1)
+
+	var averageBlockTimeMilliSecond *big.Float
+	// Determine case A or case B
+	if isTotalBlockCountExcess {
+		latestBlockHeight, err := handler.blockView.Count()
+		if err != nil {
+			return nil, fmt.Errorf("error fetching latest block height: %v", err)
+		}
+		latestBlockIdentity := block_view.BlockIdentity{MaybeHeight: &latestBlockHeight}
+		latestBlock, err := handler.blockView.FindBy(&latestBlockIdentity)
+		if err != nil {
+			return nil, fmt.Errorf("error fetching latest block: %v", err)
+		}
+
+		// Find the nth block before latest block
+		startBlockHeight := latestBlockHeight - nRecentBlocks.Int64()
+		startBlockIdentity := block_view.BlockIdentity{MaybeHeight: &startBlockHeight}
+		startBlock, err := handler.blockView.FindBy(&startBlockIdentity)
+		if err != nil {
+			return nil, fmt.Errorf("error fetching the start block: %v", err)
+		}
+
+		// Calculate total time in generating n recent blocks
+		nRecentBlocksTotalTime := latestBlock.Time.UnixNano() - startBlock.Time.UnixNano()
+
+		// Case B
+		nRecentBlocksTotalTimeMilliSecond := new(big.Float).Quo(
+			new(big.Float).SetInt64(nRecentBlocksTotalTime),
+			new(big.Float).SetInt64(int64(1000000)),
+		)
+		averageBlockTimeMilliSecond = new(big.Float).Quo(
+			nRecentBlocksTotalTimeMilliSecond,
+			new(big.Float).SetInt(nRecentBlocks),
+		)
```

Comment:
```suggestion
		// Case B
		latestBlockHeight, err := handler.blockView.Count()
		if err != nil {
			return nil, fmt.Errorf("error fetching latest block height: %v", err)
		}
		latestBlockIdentity := block_view.BlockIdentity{MaybeHeight: &latestBlockHeight}
		latestBlock, err := handler.blockView.FindBy(&latestBlockIdentity)
		if err != nil {
			return nil, fmt.Errorf("error fetching latest block: %v", err)
		}

		// Find the nth block before latest block
		startBlockHeight := latestBlockHeight - nRecentBlocks.Int64()
		startBlockIdentity := block_view.BlockIdentity{MaybeHeight: &startBlockHeight}
		startBlock, err := handler.blockView.FindBy(&startBlockIdentity)
		if err != nil {
			return nil, fmt.Errorf("error fetching the start block: %v", err)
		}

		// Calculate total time in generating n recent blocks
		nRecentBlocksTotalTime := latestBlock.Time.UnixNano() - startBlock.Time.UnixNano()

		nRecentBlocksTotalTimeMilliSecond := new(big.Float).Quo(
			new(big.Float).SetInt64(nRecentBlocksTotalTime),
			new(big.Float).SetInt64(int64(1000000)),
		)
		averageBlockTimeMilliSecond = new(big.Float).Quo(
			nRecentBlocksTotalTimeMilliSecond,
			new(big.Float).SetInt(nRecentBlocks),
		)
```

Reply:
Moving the comment case B to the beginning of the code block

Reply:
👌